app/src/main/java/com/github/se/cyrcle/ui/gambling/GamblingScreen.kt

```diff
@@ -0,0 +1,350 @@
+        }
+        2 -> { // Common segment
+          if (shouldTriggerNearMiss) {
+            // For Common, we want to get close to the left edge (near Legendary)
+            val maxOffsetLeft = angleWithinSegment
+            // Use only 5-10% of the distance to get very close to Legendary
+            randomOffset = -maxOffsetLeft * (0.90f + (Math.random() * 0.05f).toFloat())
+            Log.d(TAG, "Triggering near-miss on Common segment (close to Legendary)")
+          } else {
+            // Normal random offset within safe bounds
+            val maxOffsetLeft = angleWithinSegment
+            val maxOffsetRight = segmentAngle - angleWithinSegment
+            val safeOffset = minOf(maxOffsetLeft, maxOffsetRight) * 0.9f
+            randomOffset = (Math.random() * 2 * safeOffset - safeOffset).toFloat()
+          }
+        }
+        else -> {
+          // For other segments, use normal random offset within safe bounds
+          val maxOffsetLeft = angleWithinSegment
+          val maxOffsetRight = segmentAngle - angleWithinSegment
+          val safeOffset = minOf(maxOffsetLeft, maxOffsetRight) * 0.9f
+          randomOffset = (Math.random() * 2 * safeOffset - safeOffset).toFloat()
+        }
+      }
+
+      requiredRotation += randomOffset
+
+      // Ensure we're always rotating clockwise
+      if (requiredRotation <= 0) requiredRotation += 360
+
+      // Add 6-7 full rotations for effect
+      val fullRotations = (6..7).random() * 360
+      targetRotation = rotation - (requiredRotation + fullRotations)
+
+      Log.d(TAG, "Landing angle within segment: $angleWithinSegment")
+      Log.d(TAG, "Applied offset: $randomOffset degrees")
+      Log.d(TAG, "Required rotation: $requiredRotation degrees")
+      Log.d(TAG, "Total rotation with full spins: ${requiredRotation + fullRotations} degrees")
+      Log.d(TAG, "Near-miss triggered: $shouldTriggerNearMiss")
+
+      currentRotation = rotation
+      spinStartTime = System.currentTimeMillis()
+      spinDuration = 10000 // 4 seconds spin
+      isSpinning = true
+    }
+  }
+}
+
+@Composable
+fun GamblingScreen(onWheelViewCreated: ((WheelView) -> Unit)? = null) {
+  var wheelView by remember { mutableStateOf<WheelView?>(null) }
+
+  Box(
+      modifier = Modifier.fillMaxSize().testTag("gambling_screen"),
+      contentAlignment = Alignment.Center) {
+        AndroidView(
+            modifier = Modifier.size(300.dp).testTag("wheel_view"),
+            factory = { context ->
+              WheelView(context).also {
+                wheelView = it
+                onWheelViewCreated?.invoke(it)
+              }
+            })
+
+        Button(
+            onClick = { wheelView?.spin() },
+            modifier = Modifier.size(90.dp).clip(CircleShape).testTag("spin_button"),
+            colors =
+                ButtonDefaults.buttonColors(
+                    containerColor = androidx.compose.ui.graphics.Color.Red)) {
+              Text(
+                  "SPIN",
```

Comment:
This should be in R.string

Reply:
Is it you André ? How did you get Louis' git account ..?!